skimage/filters/_sparse.py

```diff
@@ -23,13 +22,59 @@ def _validate_window_size(axis_sizes):
             raise ValueError(msg)
 
 
-def _to_np_mode(mode):
-    """Convert padding modes from `ndi.correlate` to `np.pad`."""
-    mode_translation_dict = dict(nearest='edge', reflect='symmetric',
-                                 mirror='reflect')
-    if mode in mode_translation_dict:
-        mode = mode_translation_dict[mode]
-    return mode
+def _get_view(padded, kernel_shape, idx, val):
+    """Get a view into `padded` that is offset by `idx` and scaled by `val`.
+
+    If `padded` was created by padding the original image by `kernel_shape` as
+    in _mean_std, then the view created here will match the size of the
+    original image.
+    """
+    sl_shift = tuple([slice(c, s - (w_ - 1 - c))
+                      for c, w_, s in zip(idx, kernel_shape, padded.shape)])
+    v = padded[sl_shift]
+    if val == 1:
+        return v
+    elif val == -1:
+        return -v
```

Comment:
I suspected it may be faster, but hadn't timed it. I tried just now and it was faster on very small arrays, but the same on large arrays, so probably not important to have it as a special case

```Python
import numpy as np
a = np.ones((4096, 4096))

%timeit (-a)
22.6 ms ± 130 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)

%timeit (-1 * a)
22.6 ms ± 33.4 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)

a = np.ones((16, 16))

%timeit (-a)
400 ns ± 2.4 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

%timeit (-1 * a)
913 ns ± 3.76 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

Reply:
The same relative performance pattern occurs for CuPy as reported for NumPy above

Reply:
Coolio, let's remove that special case then?